src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml

```diff
@@ -18,40 +18,54 @@ limitations under the License.
-        <attribute side="server" code="0x02" define="ACTIVE_MODE_THRESHOLD" type="INT16U" min="0x12C" max="0xEA60" default="0xFA0" writable="false" optional="false" isNullable="false">ActiveModeThreshold</attribute>
+        <description>Client Monitoring allows for ensuring that listed clients are notified when a server is available for communication.</description>
+        <attribute side="server" code="0x00" define="IDLE_MODE_INTERVAL" type="INT32U" min="500" max="86400000" default="500" writable="false" optional="false" isNullable="false">IdleModeInterval</attribute>
+        <attribute side="server" code="0x01" define="ACTIVE_MODE_INTERVAL" type="INT32U" min="300" default="300" writable="false" optional="false" isNullable="false">ActiveModeInterval</attribute>
+        <attribute side="server" code="0x02" define="ACTIVE_MODE_THRESHOLD" type="INT16U" min="300" default="300" writable="false" optional="false" isNullable="false">ActiveModeThreshold</attribute>
 
-        <!-- TODO: Revisit the number of entries per fabric-->
-        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistration" length="4" writable="false" optional="false" isNullable="false">ExpectedClients</attribute>
+        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistrationStruct" writable="false" optional="false" isNullable="false" reportable="false">
+            <description>ExpectedClients</description>
+            <access op="read" privilege="administer"/>
+            <access modifier="fabric-scoped"/>
+        </attribute>
 
-        <command source="client" code="0x00" name="RegisterClientMonitoring" optional="false">
+        <attribute side="server" code="0x04" define="ICD_COUNTER" type="INT32U" writable="false" optional="false" isNullable="false" reportable="false">
+            <description>ICDCounter</description>
+            <access op="read" privilege="administer"/>
+        </attribute>
+
+
+        <command source="client" code="0x00" name="RegisterClientMonitoring" response="RegisterClientMonitoringResponse" isFabricScoped="true" optional="false">
             <description> Register a client to the end device </description>
             <arg name="ClientNodeId" type="node_id"/>
-            <arg name="ICid" type="INT64U"/>
+            <arg name="Key" type="OCTET_STRING" length="16"/>
             <access op="invoke" privilege="manage"/>
         </command>
 
-        <command source="client" code="0x01" name="UnregisterClientMonitoring" optional="false">
+        <command source="server" code="0x01" name="RegisterClientMonitoringResponse" optional="false" disableDefaultResponse="true">
+            <description> RegisterClientMonitoring response command </description>
+            <arg name="Status" type="ENUM8"/>
```

Comment:
Can you please file a followup to make the `status` type actually work right and use it here?  Feel free to assign it to me.

Reply:
opened #26317